mu-trees/addon/merge-addons-config.js

```diff
@@ -0,0 +1,67 @@
+/**
```

Comment:
I do not understand why this is included in `addon/` and not done completely in `index.js`, what am I missing?

Reply:
@rwjblue What is your proposal in this case? Let's see if I can understand your suggestion. 

In this proposal, the resolver looks like:

```resolver.js
     import Resolver from 'ember-resolver/resolvers/fallback';
     import buildResolverConfig from 'ember-resolver/ember-config';
     import config from '../config/environment';
     import addonsConfig from 'ember-resolver/addons-config';
     import mergeAddonsConfig from 'ember-resolver/merge-addons-config';

     let moduleConfig = buildResolverConfig(config.modulePrefix);
     mergeAddonsConfig(moduleConfig, addonsConfig);

     export default Resolver.extend({
       config: moduleConfig
     });
```

The only benefit is that users can modify the `addonsConfig` if needed to skip the addon collision exception.


Are you suggesting to perform the addon config merge within the `buildResolverConfig` function and continue using the current MU `resolver.js` like:

```js
     import Resolver from 'ember-resolver/resolvers/fallback';
     import buildResolverConfig from 'ember-resolver/ember-config';
     import config from '../config/environment';
     

     let moduleConfig = buildResolverConfig(config.modulePrefix);

     export default Resolver.extend({
       config: moduleConfig
     });
```

or something else?

Reply:
I'm suggesting to do the final merging in node land and using broccoli-file-creator to write the final config to `ember-resolver/ember-config` directly.

Reply:
@rwjblue 

I think autogenerating the whole `ember-resolver/ember-config` is not ideal, because this is a function that needs the `config.modulePrefix`.

I have changed my implementation to execute `mergeAddonsConfig` within the `ember-resolver/ember-config` which has been refactored:

```js
import addonsConfig from 'ember-resolver/addons-config';
import moduleConfig from 'ember-resolver/module-config';
import mergeAddonsConfig from 'ember-resolver/utils/merge-addons-config';

export default function generateConfig(name) {

  let result = {
    app: {
      name,
      rootName: name
    },
  };

  mergeAddonsConfig(moduleConfig, addonsConfig);

  return Object.assign(result, moduleConfig);
}
```


 This setup allows using the same default `src/resolver.js` as today.

```js
     import Resolver from 'ember-resolver/resolvers/fallback';
     import buildResolverConfig from 'ember-resolver/ember-config';
     import config from '../config/environment';
     
     let moduleConfig = buildResolverConfig(config.modulePrefix);

     export default Resolver.extend({
       config: moduleConfig
     });
```


Done at https://github.com/ember-cli/ember-resolver/pull/348/commits/4d2995021f4512e42a30b083d5f62a947f598453